tests/testthat/test-s3store.R

```diff
@@ -69,13 +68,23 @@ withr::with_options(list(
   })
 
   test_that("it does not produce an error when the object is found with an s3exists following the s3.put", {
-    map <- list2env(list("s3://test/key" = NULL))
     testthatsomemore::package_stub("base", "system2",  function(...) TRUE, {
     testthatsomemore::package_stub("s3mpi", "s3exists",  function(...) TRUE, {
     testthatsomemore::package_stub("s3mpi", "s3.put", function(...) run_system_put(..2, ..3, "", TRUE, 0), {
       expect_error(s3store("value", "key"), NA)
     })})})
   })
+
+  calling_intervals <- NULL
+  test_that("it can retry with the correct timing when an s3exists returns FALSE", {
+    testthatsomemore::package_stub("base", "Sys.sleep",  function(...) calling_intervals <<- c(calling_intervals, ..1), {
+    testthatsomemore::package_stub("base", "system2",  function(...) TRUE, {
+    testthatsomemore::package_stub("s3mpi", "s3exists",  function(...) FALSE, {
+    testthatsomemore::package_stub("s3mpi", "s3.put", function(...) run_system_put(..2, ..3, "", TRUE, 3, c(1, 2, 100)), {
+      try(run_system_put(..2, ..3, "", TRUE, 3, c(1, 2, 100)), silent = T)
```

Comment:
If it's a nested function it will catch them from the function above...
```r
R-EXP> fn <- function(...) {
+   fn2 <- function(x, y, z) {
+     print(paste(..1, ..2, ..4), sep = "-")
+   }
+   fn2(1,2,3)
+ }
R-EXP> fn(10,11,12,13)
[1] "10 11 13"
```